array.c

```diff
@@ -5637,16 +5637,18 @@ recursive_equal(VALUE ary1, VALUE ary2, int recur)
 
 /*
  *  call-seq:
- *     ary == other_ary   ->   bool
- *
- *  Equality --- Two arrays are equal if they contain the same number of
- *  elements and if each element is equal to (according to Object#==) the
- *  corresponding element in +other_ary+.
- *
- *     [ "a", "c" ]    == [ "a", "c", 7 ]     #=> false
- *     [ "a", "c", 7 ] == [ "a", "c", 7 ]     #=> true
- *     [ "a", "c", 7 ] == [ "a", "d", "f" ]   #=> false
- *
+ *    array == other_array -> true or false
+ *
+ *  Returns +true+ if both <tt>array.size == other_array.size</tt>
+ *  and for each index +i+ in +array+, <tt>array[i] == other_array[i]</tt>:
+ *    a = [0, 1, 2]
```

Comment:
The examples for `==` and `eql?` should highlight cases where behavior is different.  For example `1 == 1.0` is true, but `1.eql?(1.0)` is false.  So it would be good to show something like that in the examples.